x/bucket/keeper/grpc_query_service_for_bucket.go

```diff
@@ -0,0 +1,26 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/sonr-io/sonr/x/bucket/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) ServiceForBucket(c context.Context, req *types.QueryGetServiceForBucketRequest) (*types.QueryGetServiceForBucketResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	svc, found := k.GetService(ctx, req.Uuid)
+	if !found {
+		return nil, fmt.Errorf("error while querying service for bucket: %s %s", req.Uuid, sdkerrors.ErrKeyNotFound)
```

Comment:
I think specifying that the resource wasn't found with the identifier would be better in this context rather than saying an error occurred.